src/main/resources/static/views/project/ProjectController.js

```diff
@@ -1,4 +1,115 @@
 var ProjectController = angular.module('greatStartApp')
-    .controller('ProjectController', ['$scope', function ($scope) {
+    .controller('ProjectController', function ($rootScope, $scope, $uibModal, $routeParams, Project, $location) {
+        var investedAmount = function (project) {
+            var result = 0;
+            for (var i = 0; i < project.dtoInvestments.length; i++) {
+                if (project.dtoInvestments[i].paid) {
+                    result += project.dtoInvestments[i].sum;
+                }
+            }
+            return result;
+        };
 
-    }]);
\ No newline at end of file
+        $scope.projectImage = '';
+        $scope.projectCroppedImage = '';
+        $scope.imageChanged = false;
+
+        var handleProjectImageSelect = function (evt) {
+            var file = evt.currentTarget.files[0];
+            var reader = new FileReader();
+            reader.onload = function (evt) {
+                $scope.$apply(function ($scope) {
+                    $scope.projectImage = evt.target.result;
+                    $scope.imageChanged = true;
+                });
+            };
+
+            reader.readAsDataURL(file);
+
+        };
+
+        angular.element(document.querySelector('#fileInput')).on('change', handleProjectImageSelect);
+
+        $scope.update = function (project) {
+            return $scope.project = Project.update({id: project.id}, project, function () {
+                $scope.project = Project.get({id: project.id})
+            })
+        };
+
+        $scope.submit = function() {
+            if ($scope.imageChanged) {
+                $scope.project.desc.image = $scope.projectCroppedImage.replace(/^data:image\/[a-z]+;base64,/, "");
+            }
+            if (!$scope.project.id) {
+                $scope.createProject($scope.project);
+            } else {
+                $scope.currentProjectId = $scope.project.id;
+                $scope.update($scope.project).$promise.then(function (success) {
+                    $location.path('/project/' + $scope.currentProjectId);
+                }, function (error) {
+                    $scope.error = error.status + " " + error.statusText;
+                });
+            }
+        };
+
+        var allProjects = function() {
+            return Project.query();
+        };
+
+        if ($routeParams.id) {
+            $scope.project = Project.get({id: $routeParams.id});
+        } else {
+            $scope.project = {};
+            $scope.projects = allProjects();
+        }
+
+
+        var ivnProgressWithWidth = function (investedAmount, project) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "ivnProgressWithWidth" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of the unused 'ivnProgressWithWidth' variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AUnusedVariable)